templates/terraform/constants/apigateway_apiconfig.erb

```diff
@@ -0,0 +1,36 @@
+func apiConfigCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+    field := "gateway_config.0.backend_config.0.google_service_account"
+	old, new := diff.GetChange(field)
+	config := meta.(*Config)
+    log.Printf("[DEBUG] Running CustomizeDiff. old value: %s, new value: %s", old.(string), new.(string))
+
+	project, err := getProjectFromDiff(diff, config)
+	if err != nil {
+		return err
+	}
+
+    // We will check for diffs in 3 scenarios.
+    // 1. Matches the project's default compute engine service account
+    // 2. RegexCompile against serviceaccount self link
+    // 3. UUID of svc matches up.
+
+	serviceAccount := regexp.MustCompile("projects/-/serviceAccounts/(.+)")
+
+	if serviceAccount.MatchString(new.(string)) {
+		parts := serviceAccount.FindStringSubmatch(new.(string))
+        if old == "" {
+            // Service account was initially null and the resource used the default service account.
```

Comment:
Just to confirm- if you don't set a service account, does the API return a value for it?

Reply:
I'm currently testing it without one and it returns the value of the default service account.

Reply:
In that case, could it just be an optional+computed field instead of needing the customizediff?

Reply:
Ah, I applied it to the parent field and it works now.